pkg/operator/ceph/object/objectstore.go

```diff
@@ -347,20 +350,21 @@ func GetRealmKeyArgs(ctx context.Context, clusterdContext *clusterd.Context, rea
 	return GetRealmKeyArgsFromSecret(secret, realmNsName)
 }
 
-func getZoneEndpoints(objContext *Context, serviceEndpoint string) ([]string, error) {
+func getZoneEndpoints(objContext *Context, serviceEndpoint string) ([]string, bool, error) {
```

Comment:
Shouldn't this check just be checking the CRs? I thought the endpoints list is either what's defined in the zone CR or what's defined in the object-stores CR if there's nothing in the zone CR. So if neither of those are set in the CR we use the service endpoint.

Reply:
@alimaredia: I have modified this function to make it effective for updation, if the endpoint is already listed why do we need to call modify the zone endpoint info. Similar we need to remove the endpoint if it is present in zone

Reply:
You bring up a great point about not modifying the zone endpoint info if it is the same after an update. 

@BlaineEXE  Is there some other internal structure that we could just store the most up to date endpoints list from a CRs in? Do you think we should call `radosgw-admin zonegroup get` to get the most up to date endpoints list?

Reply:
Actually I forgot about the default service endpoint case. disregard my comments here.